src/dune_rules/coq_rules.ml

```diff
@@ -56,8 +56,29 @@ let resolve_program sctx ~loc ~dir prog =
     ~hint:"opam install coq"
 
 module Coq_plugin = struct
+  let meta_info ~context (lib : Lib.t) =
+    let debug = false in
+    let name = Lib.name lib |> Lib_name.to_string in
+    if debug then Format.eprintf "Meta info for %s@\n" name;
+    match Lib_info.status (Lib.info lib) with
+    | Public (_, pkg) ->
+      let package = Package.name pkg in
+      let meta_i =
+        Path.Build.relative
+          (Local_install_path.lib_dir ~context ~package)
+          "META"
+      in
+      if debug then
+        Format.eprintf "Meta for %s: %s@\n" name (Path.Build.to_string meta_i);
+      Some (Path.build meta_i)
+    | Installed -> None
+    | Installed_private | Private _ ->
+      (* XXX: Do a warning here? Or maybe guard on the Dune lang version ? *)
+      User_error.raise
```

Comment:
There is a similar error in coq_lib that you may be able to use.

Reply:
I made it warning a bit different so I couldn't reuse it, but not a big deal IMHO.